### ISSUE TITLE
Changes in .bib not reflected until restart of Jekyll

### ISSUE THREAD

**Issue by reitzig** [inukshuk/jekyll-scholar#104]
When running `jekyll serve` (e.g. locally for development), changes in source files prompt a rebuild so that the effect can be viewed in the browser (virtually) live.

Changes to `. bib` files, however, are *not* reflected in the output; you have to restart `jekyll` every time you change them.

Please change this so that `jekyll-scholar` picks up on changing `.bib` files.

**Comment by inukshuk** [comment]
@mamoll do you know if this behaviour is changed in jekyll 3 using your additions? It would seem reasonable if the reload and cache logic used the same files, so this might be addressed already (though I'm just guessing here).

**Comment by mamoll** [comment]
I cannot reproduce this. I suspect that @reitzig is not using the prerelease. These are the jekyll-related gems I have installed:
```
jekyll (3.0.0.pre.beta8, 2.5.3)
jekyll-coffeescript (1.0.1)
jekyll-gist (1.3.4, 1.3.3)
jekyll-less (0.0.4)
jekyll-minifier (0.0.5.pre, 0.0.4)
jekyll-paginate (1.1.0)
jekyll-redirect-from (0.8.0)
jekyll-sass-converter (1.3.0)
jekyll-scholar (5.0.0.pre.1)
jekyll-watch (1.2.1)
```

**Comment by reitzig** [comment]
Right, I'm on the current stable versions, as far as I know.

```
jekyll (2.4.0)
jekyll-coffeescript (1.0.1)
jekyll-feed (0.3.1)
jekyll-gist (1.3.4)
jekyll-mentions (0.2.1)
jekyll-paginate (1.1.0)
jekyll-redirect-from (0.8.0)
jekyll-sass-converter (1.3.0)
jekyll-scholar (4.5.0)
jekyll-sitemap (0.8.1)
jekyll-watch (1.2.1)
```

**Comment by inukshuk** [comment]
It is likely that this issue is already addressed by @mamoll's -- they will be released when jekyll 3 is out. If you want to try before that you need to install the jekyll 3 beta and use the jekyll-scholar master branch.

**Comment by reitzig** [comment]
@inukshuk With 5.1.0 I notice that bibliographies get updated when the BibTeX changes but details pages do not. Has that changed since and an update to the most recent version helps? Or should I open a new issue?

**Comment by inukshuk** [comment]
Do you know if Jekyll's reload cache support page generators as well? I haven't used Jekyll in a while, so I'm not up to speed with their latest features. If page generators are support this in principle then we should support it, too.

**Comment by mamoll** [comment]
@inukshuk I don't use the details pages. Most likely you need another change like this: 5b2fbf05b6f48d9c6f2490a6ab9e99075c3eb275 for detail pages.

**Comment by reitzig** [comment]
@inukshuk I have no idea, sorry.

**Comment by inukshuk** [comment]
@mamoll thanks! I added this to the page generator though I'm not sure if this is supposed to work for generated pages -- @reitzig can you try?

**Comment by inukshuk** [comment]
I think it might work if you make changes to an existing entry; but if you add a new entry this cannot really work (since the corresponding page does not exist) and likewise for removing entries.

**Comment by reitzig** [comment]
Thanks, but I don't have the time to tinker with dev branches on top of everything. Once it's in a gem update I can pull I'll happily test!

If changes to existing entries are caught, that's already a huge help. I use the `note` field to give more flavour to the details pages (arbitrary Markdown and HTML in there!) so seeing these updates without having to restart the server every time is a huge boon.

Obviously, new/missing entries should ideally be caught, but that happens far less frequently so I don't estimate it to be a huge annoyance. Confusing, for sure.

**Comment by inukshuk** [comment]
Landed in 5.4.1

**Comment by reitzig** [comment]
Thanks!
